pkg/controllers/queue/queue_controller_action.go

```diff
@@ -82,24 +100,72 @@ func (c *queuecontroller) syncQueue(queue *schedulingv1beta1.Queue, updateStateF
 		queueStatus.Allocated = v1.ResourceList{}
 	}
 
+	newQueue := queue.DeepCopy()
 	// ignore update when status does not change
-	if equality.Semantic.DeepEqual(queueStatus, queue.Status) {
+	if !equality.Semantic.DeepEqual(queueStatus, queue.Status) {
+		newQueue.Status = queueStatus
+		var err error
+		newQueue, err = c.vcClient.SchedulingV1beta1().Queues().UpdateStatus(context.TODO(), newQueue, metav1.UpdateOptions{})
+		if err != nil {
+			klog.Errorf("Failed to update status of Queue %s: %v.", newQueue.Name, err)
+			return err
+		}
+	}
+
+	if newQueue.Name == "root" {
 		return nil
 	}
 
-	newQueue := queue.DeepCopy()
-	newQueue.Status = queueStatus
-	if _, err := c.vcClient.SchedulingV1beta1().Queues().UpdateStatus(context.TODO(), newQueue, metav1.UpdateOptions{}); err != nil {
-		klog.Errorf("Failed to update status of Queue %s: %v.", newQueue.Name, err)
+	parentQueue, err := c.queueLister.Get(newQueue.Spec.Parent)
+	if err != nil {
+		klog.Errorf("Failed to get parent queue of Queue %s: %v.", newQueue.Name, err)
 		return err
 	}
 
+	switch parentQueue.Status.State {
+	case schedulingv1beta1.QueueStateClosed, schedulingv1beta1.QueueStateClosing:
+		if newQueue.Status.State != schedulingv1beta1.QueueStateClosed && newQueue.Status.State != schedulingv1beta1.QueueStateClosing {
+			_, err = c.updateQueueAnnotation(newQueue, ClosedByParentAnnotationKey, ClosedByParentAnnotationTrueValue)
+			if err != nil {
+				klog.Errorf("Failed to patch annotation of Queue %s: %v.", newQueue.Name, err)
+				return err
+			}
+
+			req := &apis.Request{
+				QueueName: newQueue.Name,
+				Action:    busv1alpha1.CloseQueueAction,
+			}
+
+			c.enqueue(req)
```

Comment:
Why re-enqueue here？

Reply:
The main reason for re-enqueuing here is that the parent queue of the current queue is in a closed or closing state, while the current queue itself is not in a closed or closing state. Therefore, the purpose of re-enqueuing here is to close the queue.